File: 영진/2026.05.20/연속된 부분 수열의 합.cpp
#include<bits/stdc++.h>
using namespace std;

vector<int> solution(vector<int> sequence, int k) {
        vector<int> answer;
        int m = sequence.size();
    
        int size = sequence.size();
        
        int start = 0;
        int end = 0;
        int sum =sequence[0];
    
        while(end<size){
            if(sum == k){
                // 더 작은 부분수열을 찾은 경우
                // 크기가 같으면 갱신 안 함
                if(end-start < m){
                    m = end-start;
                    answer= {start,end};
                    // if(m == 0) return answer;
                }
                // 더 나은 경우의 수를 찾기 위해 start++
                sum -= sequence[start];
                start++;
            }else if(sum > k){
                sum -= sequence[start];
                start++;
            }else{ // sum < k
                end++;
                if(end<size) sum += sequence[end];
            }
        }
    
    return answer;
}

File: 현지/2026_05_20/연속된 부분 수열의 합.py
def solution(sequence, k):

    s = 0
    e = 0

    total = sequence[0]

    answer = [0, len(sequence)-1]

    while True:

        if total < k:

            e += 1

            if e >= len(sequence):
                break

            total = total + sequence[e]

        elif total > k:

            total = total - sequence[s]

            s += 1

        else:

            if (answer[1] - answer[0]) > (e - s):
                answer = [s, e]

            total -= sequence[s]

            s = s + 1

    return answer
